application/transports/wrapping/obfs4/client.go

```diff
@@ -14,6 +20,8 @@ import (
 // the station side Transport struct has one instance to be re-used for all sessions.
 type ClientTransport struct {
 	Parameters *pb.GenericTransportParams
+	connectTag []byte
+	keys       dd.ConjureSharedKeys
```

Comment:
We should be able to kill two birds with one stone if we make the correct changes here wrt. the `Obfs4Keys`.

We do not need the whole `ConjureSharedKeys` object to be attached to a client Transport instance (also it brings the `import dd "...` . And outside of a client transport `Obfs4Keys` have no use. 

So to me what makes sense would be to remove `ObfsKeys` from the `ConjureSharedKeys` struct and move it and everything relating to its generation into the obfs4 transport (Here). Specifically generating the Obfs4Keys for a client transport instance only when `PrepareKeys` is called.

There is a little more to this than there seems, because we need an `io.Reader` for our psuedo-random key. But we don't just want to regenerate our hkdf, because that would mean that we might use the same bytes from the hkdf in different places (and if one of those places goes out over the wire, then you have just sent your private key in plaintext). Plus the station has to be able to generate the same key so we need to be consistent about how many bytes have been read from the hkdf pseudo-random `io.Reader`.

For now what I think satisfies this is to add an `io.Reader` to the `ConjureSharedKeys` struct and keep the hkdf that we build when `generateSharedKeys` gets run so that when we call `PrepareKeys` we are not re-reading bytes from the prng.

https://github.com/refraction-networking/gotapdance/blob/2949afdb05b2da676ced22b8d3f59cd8e1c8cb7c/tapdance/conjure.go#L924-L954